includes/core/class-wc-payments-order.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Class file for WCPay\Core\WC_Payments_Order
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core;
+
+defined( 'ABSPATH' ) || exit;
+
+use \WC_Order;
+use WC_Payments_Utils;
+
+/**
+ * WC_Payments_Order
+ */
+class WC_Payments_Order {
+
+	/**
+	 * The WC_Order corresponding to this WC_Payments_Order.
+	 *
+	 * @var WC_Order
+	 */
+	private $order;
+
+	/**
+	 * Constructor when order_id is provided.
+	 *
+	 * @param WC_Order|int $order The order Id.
+	 */
+	public function __construct( $order ) {
+		$this->order = wc_get_order( $order );
```

Comment:
`wc_get_order` always constructs a new object, and I'm not sure we want that.

- On one hand it's good, because there will probably be no side effects.
- On the other hand, imagine this:

```php
$order = 'A WC_Order object, which we already have access to.';
$wcpay_order = new WC_Payments_Order( $order );
$wcpay_order->set_payment_intent_id( 'pi_XXX' ); // this is a public method, this can easily be done.
// Since there is no `save()` method of `WC_Payments_Order`, the only option is to save the original.
$order->save();
```

But this would accomplish nothing, since `WC_Payments_Order` only modified its internal instance, which was not saved. This example looks ugly though, as we're creating a second object, and saving the first one.

IMO we should require an `int $order_id` here, to make it obvious, that the received `$order` object will not be modified at any stage. If someone wants a shortcut, we could create a static `WC_Payments_Order::from_wc_order()` method.

Reply:
Changed to int `$order_id`